install.sh

```diff
@@ -0,0 +1,220 @@
+        p)
+            PYTHON_ONLY=1;;
+        b)
+            JUST_BOT=1;;
+        l)
+            BOT_AND_PYTHON=1;;
+        # if a bad argument is given, then throw an error
+        \?)
+            echo "Invalid option: -$OPTARG" >&2 Help exit 1;;
+        :)
+            echo "Option -$OPTARG requires an argument." >&2 Help exit 1;;
+    esac
+done 
+
+# Install dependencies for MacOS
+function install_macos(){
+    # Check if homebrew is installed
+    if [ ! command -v brew &> /dev/null ]; then
+        echo "Installing Homebrew"
+        # if it's is not installed, then install it in a NONINTERACTIVE way
+        NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)" 
+        # Check for what arcitecture, so you can place path.
+        if [[ "uname -m" == "x86_64" ]]; then
+            echo "export PATH=/usr/local/bin:$PATH" >> ~/.bash_profile && source ~/.bash_profile
+        fi
+    # If not
+    else
+        # Print that it's already installed
+        echo "Homebrew is already installed"
+    fi
+    # Install the required packages
+    echo "Installing required Packages" 
+    brew install python@3.10 tcl-tk python-tk
+} 
+
+# Function to install for arch (and other forks like manjaro)
+function install_arch(){ 
+    echo "Installing required packages"
+    sudo pacman -S --needed python3 tk git && python3 -m ensurepip || install_fail
+} 
+
+# Function to install for debian (and ubuntu)
+function install_deb(){ 
+    echo "Installing required packages"
+    sudo apt install python3 python3-dev python3-tk python3-pip git || install_fail
+} 
+
+# Function to install for fedora (and other forks)
+function install_fedora(){ 
+    echo "Installing required packages" 
+    sudo dnf install python3 python3-tkinter python3-pip git python3-devel || install_fail
+} 
+
+# Function to install for centos (and other forks based on it)
+function install_centos(){
+    echo "Installing required packages"
+    sudo yum install -y python3 || install_fail
+    sudo yum install -y python3-tkinter epel-release python3-pip git || install_fail
+} 
+
+function get_the_bot(){ 
+    echo "Downloading the bot" 
+    git clone https://github.com/elebumm/RedditVideoMakerBot.git 
+} 
+
+#install python dependencies
+function install_python_dep(){ 
+    # tell the user that the script is going to install the python dependencies
+    echo "Installing python dependencies" 
+    # cd into the directory
+    cd RedditVideoMakerBot 
+    # install the dependencies
+    pip3 install -r requirements.txt 
+    # cd out
+    cd ..
+} 
+
+# install playwright function
+function install_playwright(){
+    # tell the user that the script is going to install playwright 
+    echo "Installing playwright"
+    # cd into the directory where the script is downloaded
+    cd RedditVideoMakerBot
+    # run the install script
+    playwright install 
+    playwright install-deps 
```

Comment:
These should be ran as `python -m` by default, it's highly unlikely the user will have playwright installed on their system

Reply:
Playwright is installed with requirements.txt, but I will make the change

Reply:
Yh I meant by `pacman` or `apt`, all good